### ISSUE TITLE
ArgumentError: wrong number of arguments (given 1, expected 2)

### ISSUE THREAD

**Issue by HoneyryderChuck** [oracle/truffleruby#2111]
Coming from https://github.com/oracle/truffleruby/issues/2103 .

I've tracking support for truffleruby in `httpx`, where the build is failing with the error from the title.

Link to a failing build: https://gitlab.com/honeyryderchuck/httpx/-/jobs/770843884

The error happens in a pattern calling super like this:

```ruby
      class << self
        def new(*, options)
          return options.body if options.body.is_a?(self)

          super # blow up here
        end
      end
```

**Comment by aardvark179** [comment]
I can reproduce the issue and can see the cause of the problem. We don't correctly handle named arguments given after a splat argument when copying everything we need for the super call - we may be missing some other corner cases as well. I'll take a look at fixing it.

**Comment by norswap** [comment]
@aardvark179 I took a stab at clarifying what goes on with argument-to-parameter mapping back in 2017 (so might be outdated): https://norswap.com/ruby-methods/
Might be useful to catch some of these corner cases.

**Comment by eregon** [comment]
Fixed in 47ec1d30d12c29227fd22575b26f0fd83665b227 by @aardvark179
